ir/type.cpp

```diff
@@ -247,10 +252,87 @@ void VectorType::print(ostream &os) const {
   os << "TODO";
 }
 
+expr AggregateType::getTypeConstraints() const {
+  if (childrenType.size() < 2)
+    return false;
+
+  expr res(true);
+  for (auto &type : childrenType) {
+    res &= type->enforceIntType();
+  }
+
+  return res;
+}
+
+expr AggregateType::operator==(const AggregateType &rhs) const {
+  if (childrenType.size() != rhs.childrenType.size())
+    return false;
+
+  expr res(false);
+  for (unsigned i = 0; i < childrenType.size(); i++) {
+    res = res || *childrenType[i] == *rhs.childrenType[i];
+  }
+
+  return res;
+}
+
+expr AggregateType::enforceAggregateType() const {
+  return !childrenType.empty();
+}
+
+void AggregateType::fixup(const Model &m) {
+  // TODO
+}
+
+expr AggregateType::enforceIntOrPtrOrVectorType() const {
+  return false;
+}
+
+unsigned AggregateType::bits() const {
+  unsigned res = 0;
+  for (unsigned i = 0; i < childrenType.size(); i++) {
+    res += childrenType[i]->bits();
+  }
+
+  return res;
+}
+
+void AggregateType::print(ostream &os) const {
+  assert(childrenType.size() > 0);
+  os << '{';
+  childrenType[0]->print(os);
+  for (unsigned i = 1; i < childrenType.size(); i++) {
+    os << ", ";
+    childrenType[i]->print(os);
+  }
+  os << '}';
+}
+
+expr AggregateType::getChildConstraints(Type &type, unsigned index) const {
```

Comment:
This method is strange. I thing something that returns the type of element x is sufficient.